pkg/compat/client.go

```diff
@@ -151,6 +164,10 @@ func (c Client) Create(ctx context.Context, in runtime.Object) error {
 // The resource will be converted to a compatible version as needed.
 func (c Client) Delete(ctx context.Context, in runtime.Object, opt ...k8sclient.DeleteOptionFunc) error {
 	obj := c.downConvert(in)
+	err := scheme.Scheme.Convert(in, obj, ctx)
```

Comment:
For clarity & consistency, let's do the _down-conversion_ in the `downConvert()` method.

Reply:
👍